pkgs/development/libraries/libvirt/default.nix

```diff
@@ -66,45 +66,46 @@ in stdenv.mkDerivation rec {
     substituteInPlace src/lxc/lxc_conf.c \
       --replace 'lxc_path,' '"/run/libvirt/nix-emulators/libvirt_lxc",'
     patchShebangs . # fixes /usr/bin/python references
-    mkdir -p build && cd build
   '';
 
-  configureScript = "../configure";
-
   dontAddDisableDepTrack = true;
 
-  configureFlags = [
-    "--with-runstatedir=/run" # TODO: remove when autoconf 2.70 is released
-    "--localstatedir=/var"
-    "--sysconfdir=/var/lib"
-    "--with-libpcap"
-    "--with-qemu"
-    "--with-vmware"
-    "--with-vbox"
-    "--with-test"
-    "--with-esx"
-    "--with-remote"
-    "--with-polkit"
+  mesonFlags = let
+    opt = option: enable: "-D${option}=${if enable then "enabled" else "disabled"}";
+  in [
+    "-Dauto_features=disabled"
+    "-Drunstatedir=/run"
+    "-Dlocalstatedir=/var"
+    "-Dsysconfdir=/var/lib"
+    "-Dlibpcap=enabled"
+    "-Ddriver_qemu=enabled"
+    "-Ddriver_vmware=enabled"
+    "-Ddriver_vbox=enabled"
+    "-Ddriver_test=enabled"
+    "-Ddriver_esx=enabled"
+    "-Ddriver_remote=enabled"
+    "-Dpolkit=enabled"
+    "-Dbus=enabled"
+    "-Dfuse=disabled" # todo check against previous config
+    "-Dhal=disabled" # todo check against previous config
+    "-Dglusterfs=disabled" # todo check against previous config
+    (opt "storage_iscsi" enableIscsi)
+    (opt "storage_zfs" (stdenv.isLinux && zfs != null))
   ] ++ optionals stdenv.isLinux [
-    "QEMU_BRIDGE_HELPER=/run/wrappers/bin/qemu-bridge-helper"
-    "QEMU_PR_HELPER=/run/libvirt/nix-helpers/qemu-pr-helper"
-    "EBTABLES_PATH=${ebtables}/bin/ebtables-legacy"
-    "CFLAGS=-I${libtirpc.dev}/include/tirpc"
-    "--with-attr"
-    "--with-apparmor"
-    "--with-secdriver-apparmor"
-    "--with-numad"
-    "--with-macvtap"
-    "--with-virtualport"
-    "--with-storage-disk"
-  ] ++ optionals (stdenv.isLinux && zfs != null) [
-    "--with-storage-zfs"
-  ] ++ optionals enableIscsi [
-    "--with-storage-iscsi"
-  ] ++ optionals enableCeph [
-    "--with-storage-rbd"
+    # "QEMU_BRIDGE_HELPER=/run/wrappers/bin/qemu-bridge-helper"
+    # "QEMU_PR_HELPER=/run/libvirt/nix-helpers/qemu-pr-helper"
+    # "EBTABLES_PATH=${ebtables}/bin/ebtables-legacy"
+    # "CFLAGS=-I${libtirpc.dev}/include/tirpc"
+    "-Dattr=enabled"
+    "-Dapparmor=enabled"
+    "-Dsecdriver_apparmor=enabled"
+    "-Dnumad=enabled"
+    "-Dmacvtap=enabled"
+    "-Dvirtualport=enabled"
+    "-Dstorage_disk=enabled"
+    (opt "storage_rbd" enableCeph)
   ] ++ optionals stdenv.isDarwin [
-    "--with-init-script=none"
+    "-Dinit_script=none"
   ];
 
   installFlags = [
```

Comment:
`installFlags` will not do anything – ninja does not allow install time configuration.